enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,470 @@
+- Replace and deprecate the current placementrule APIs in https://github.com/open-cluster-management/multicloud-operators-placementrule
+
+### Non-Goals
+
+- Support tolerations/taints. ManagedCluster has no taint defined in spec yet;
+- Create metrics data when creating/updating placement decisions. It needs further investigation and should be covered in a separate enhancement; 
+
+## Proposal
+
+### User Stories
+
+#### Story 1: A cluster-admin on hub can place a policy or workload to all clusters in all ManagedClusterSets; The cluster-admin on hub can also place a policy or workload to any selected clusters in ManagedClusterSets.
+
+#### Story 2: A normal user on hub can place a policy or a workload to a slice of clusters in one or multiple ManagedClusterSets that the user is authorized to access.
+
+#### Story 3: A user can use predicate to select clusters
+  - Select openshift clusters only
+  - Select clusters in aws and gcp
+  - Select cluster not in vsphere
+  - Select openshift clusters with a specific version
+  - Select clusters labeled as prod (which is set by users).
+
+#### Story 4: A user can define the number of clusters to be selected
+  - Select 3 openshift cluster
+  - Select 5 cluster on aws
+
+#### Story 5: The placement API should be scalable to support selecting hundreds of clusters from thousands of clusters.
+
+#### Story 6: A user can select clusters across different regions.
+
+#### Story 7: A controller is able to consume placements to select desired clusters.
+
+### API Spec
+
+#### Placement API
+`Placement` is a namespace scoped API and represents a selection of clusters in one or multiple ManagedClusterSets. When a placement is created, a controller consuming the placement should generate one or multiple placement decisions based on the spec defined in placement.
+```go
+// Placement defines a rule to select a set of ManagedClusters from the ManagedClusterSets bound
+// to the placement namespace.
+//
+// Here is how the placement policy combines with other selection methods to determine a matching
+// list of clusters:
+// 1) Clusters are registered with hub as cluster-scoped ManagedClusters;
+// 2) ManagedClusters are organized into cluster-scoped ManagedClusterSets;
+// 3) ManagedClusterSets are bound to workload namespaces;
+// 4) Namespace-scoped Placements specify a slice of ManagedClusterSets which select a working set
+//    of potential ManagedClusters;
+// 5) Then Placements subselect from that working set using name/label/claim selection.
+//
+// No ManagedClusters will be selected if there is no ManagedClusterSet bound to the placement
+// namespace. User is able to bind a ManagedClusterSet to a namespace in two ways:
+// 1) Create a ManagedClusterSetBinding in the namespace if they have an RBAC rule to CREATE
+//    on the virtual subresource of `managedclustersets/bind`;
+// 2) Create a service account with name 'placement', and create a ClusterRoleBinding to grant
```

Comment:
Since permissions could be created by edits to roles, clusterroles, clusterolebindings, rolebindings, external webhooks, how will a controller know that it has to refresh the placementdecisions?

Reply:
The ManagedClusterSetBindings will be created automatically according to RBAC rules of service account `placement`. And any changes on ManagedClusterSetBindings in the placement namespace will trigger the refresh of placementdecisions. I'll remove way 2 from the api doc because it is a separate logic and has nothing to do with the placement API itself. And we may not need it at all.